### PR TITLE
[REVIEW] Fix duplicate names issue in `MultiIndex.deserialize `

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #9254 

This PR fixes `deserialize` in `cudf.MultiIndex` so that there is no data-corruption happening when there are duplicate names.

### CODE REVIEW

File: python/cudf/cudf/tests/test_multiindex.py [hunk 1]
```diff
@@ -1553,3 +1555,32 @@ def test_multiIndex_duplicate_names():
...
+            "four": [0.2, 0.1, -10.2],
+        }
+    )
+    actual_df = df.set_index(["one", "two", "three"])
+    actual_df.index.names = names
+    local_file = BytesIO()
+
+    pickle.dump(actual_df, local_file)
+    local_file.seek(0)
+    expected_df = pickle.load(local_file)
```
Comment (vyasr) [inline python/cudf/cudf/tests/test_multiindex.py:1585]: I think you've switched the expected and actual here?

**Review by vyasr — APPROVED** [review]
Fix looks good, one minor suggested improvement to the tests.

File: python/cudf/cudf/tests/test_multiindex.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_multiindex.py]: Fixed it 👍

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**